influxql/parse_tree.go

```diff
@@ -145,6 +155,11 @@ func init() {
 			return p.parseShowSubscriptionsStatement()
 		})
 		show.Group(TAG).With(func(tag *ParseTree) {
+			tag.Group(KEY).With(func(key *ParseTree) {
```

Comment:
You can do `tag.Group(KEY).Handle(CARDINALITY, ...)` instead and this will be more concise. A similar comment on the other references too. You only need `With()` if it is going to branch in multiple directions.